Sprint_1

Ruurdbk opened the pull request:
Sprint 1 of script to be accepted and merged by client to the Master branch.

Ruurdbk committed:
Sprint_1

Sprint 1 of script to be accepted and merged by client to the Master branch.

boelek commented:
Hi Ruurd, Thanks for your sprint 1 code. Looks fine: clear coding with good use of comments. We were able to run your code in our environment and the resulting .xyz file can be read by Foruno map plotters :-). So from a functionality point of view, your code meets the design specs.

Two requests if you have time left for your in your sprint 01:

1) could you see if you can rewrite the code so it makes use of functions? I would suggest a "convertNED2XYZ function that takes the arguments filename, inputdir and outputdir. Here is an example of such a function:
def convertNed2Xyz( name, in_file, out_file):
    "convert ned to xyz)"
    #open input file
    in_file_obj = open(in_file+name,"r")
    #create output file
    name=name[:-3]+'xyz'
    out_file_obj = open(out_file+name, "w")
    
    for line in in_file_obj:
        ned_sounding = line.split(in_delimiter)
        xyz_sounding = ned_sounding[1]+out_delimiter+ned_sounding[0]+out_delimiter+"-"+ned_sounding[2]
        out_file_obj.write(xyz_sounding)
        
    #close files 
    in_file_obj.close()
    out_file_obj.close()
    return

2) Could you make it so that the code reads all the NED files in a particular folder? The functionality for this can be find in the Python library called "os". Again, see if you can do this in a function. Here is an example:
def find_ned( in_path ):
   "This finds the .NED files in the directory"
   files = os.listdir( in_path )
   ned = []
   for file in files:
      if file.endswith(".NED"):
         ned.append(file)
   return ned

Nice work 👍

Ruurdbk committed:
Sprint_1 Update_01

This code now makes use of *ConvertNed2Xyz* and *find_ned* functions and reads the NED files in directories. The script has been tested and includes all requirements and improvements which the client requested. :)

Ruurdbk commented:
A new commit has been created in the sprint_1 branch. With your permission I will merge the branch as the final sprint_1 version.

boelek commented:
Very nice. Runs like a charm. We have already converted a batch. So I consider sprint 1 finished. The functionality is in place and the code is clear with the new functions. Thanks. 
🥇
Looking forward to the efficiency stats in sprint 2
